test/hotspot/jtreg/runtime/os/TestExplicitPageAllocation.java

```diff
@@ -0,0 +1,207 @@
+
+    }
+
+
+    private static int requiredPageCount(int index) {
+        int pageSizeInKB =  1 << (index-10);
+        return HEAP_SIZE_IN_KB/pageSizeInKB;
+    }
+
+    private static String sizeFromIndex(int index) {
+        int k = 1024;
+        int m = 1024*1024;
+        int g = 1024*1024*1024;
+        int sizeInBytes = 1 << index;
+        if (sizeInBytes < m)
+           return Integer.toString(sizeInBytes/k)+"K";
+        if(sizeInBytes < g)
+           return Integer.toString(sizeInBytes/m)+"M";
+        else
+           return Integer.toString(sizeInBytes/g)+"G";
+    }
+
+    private static boolean checkOutput(OutputAnalyzer out, String pageSize) throws Exception {
+        List<String> lines = out.asLines();
+        for (int i = 0; i < lines.size(); ++i) {
+            String line = lines.get(i);
+            System.out.println(line);
+            if (HEAP_PATTERN.matcher(line).find()) {
+                Matcher trace = PAGE_SIZE_PATTERN.matcher(line);
+                trace.find();
+                String tracePageSize = trace.group(1);
+                if (pageSize.contains(tracePageSize)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public static int checkAndReadFile(String filename, String pageSize) throws Exception {
+        fis = new FileInputStream(filename);
+        dis = new DataInputStream(fis);
+        int pagecount = Integer.parseInt(dis.readLine());
+        dis.close();
+        fis.close();
+        return pagecount;
+    }
+
+    public static void doSetup() throws Exception {
+        // Large page sizes
+        File[] directories = new File(DIR_HUGE_PAGES).listFiles(File::isDirectory);
+        for (File dir : directories) {
+            String pageSizeFileName = dir.getName();
+            Matcher matcher = HUGEPAGE_PATTERN.matcher(pageSizeFileName);
+            matcher.find();
+            String pageSize = matcher.group(1);
+
+            if (pageSize != null) {
+                int freePageCount = checkAndReadFile(DIR_HUGE_PAGES+pageSizeFileName+"/free_hugepages", pageSize);
+                int resvPageCount = checkAndReadFile(DIR_HUGE_PAGES+pageSizeFileName+"/resv_hugepages", pageSize);
+
+                int availablePages = freePageCount - resvPageCount;
+                if (availablePages >= 0) {
+                    System.out.println("Number of available "+pageSize+"kB pages = "+availablePages);
+                } else {
+                    System.out.println("System does not support"+pageSize+"kB pages");
+                    continue;
+                }
+
+                int index = Integer.numberOfTrailingZeros(Integer.parseInt(pageSize)*1024);
+                pageSizes[index] = true;
+                pageCount[index] = availablePages;
+            }
+        }
+        // OS vm page size
+        ProcessBuilder processBuilder = new ProcessBuilder();
+        processBuilder.command("getconf", "PAGE_SIZE");
```

Comment:
You could use the WhiteBox API to get the default page size to avoid having to start the extra process. You can search the code for `getVMPageSize()` to see other uses.